### DISCUSSION TITLE
No download info given for 'googletest' and its source directory

### DISCUSSION THREAD

**Discussion by shqln** [nasa/fprime#1368] [Q&A]
I got this error when running fprime-util generate --ut, any ideas on how to fix it? I already tried reinstalling gtest but that did not solve it.

`CMake Error at /home/danny/.local/lib/python3.6/site-packages/cmake/data/share/cmake-3.22/Modules/ExternalProject.cmake:2882 (message):
-- Configuring incomplete, errors occurred!
  No download info given for 'googletest' and its source directory:

   /home/danny/fprime/gtest/googletest-src

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
See also "/home/danny/fprime/build-fprime-automatic-native-ut/googletest-download/CMakeFiles/CMakeOutput.log".
  /home/danny/.local/lib/python3.6/site-packages/cmake/data/share/cmake-3.22/Modules/ExternalProject.cmake:3716 (_ep_add_download_command)
-- Configuring incomplete, errors occurred!
  CMakeLists.txt:6 (ExternalProject_Add)

CMake Error at cmake/googletest-download/googletest.cmake:8 (message):
  CMake step for googletest failed: 1
See also "/home/danny/fprime/build-fprime-automatic-native-ut/CMakeFiles/CMakeOutput.log".
Call Stack (most recent call first):
See also "/home/danny/fprime/build-fprime-automatic-native-ut/CMakeFiles/CMakeError.log".
  cmake/FPrime-Code.cmake:13 (include)
  CMakeLists.txt:60 (include)`

**Comment by LeStarch** [comment]
You could try removing the google test directory and try again:

```
rm -r /home/danny/fprime/gtest
fprime-util generate --ut
```

I'll admit, I have not seen this before, so I am guessing at a solution at this point.

**Reply by NishqR** [reply]
The solution here worked for us - https://stackoverflow.com/a/52392930

Had to basically install gtest and copy it from the src folder into the fprime googletest-src directory and make it again

```
sudo apt install libgtest-dev build-essential cmake
cd /usr/src/googletest
sudo cp -R * <parent_dir>/fprime/gtest/googletest-src/
cd <parent_dir>/fprime/gtest/googletest-src/
sudo cmake .
sudo cmake --build . --target install
```
